packages/drivers/odsp-driver/src/test/epochTestsWithRedemption.spec.ts

```diff
@@ -62,14 +62,13 @@ describe("Tests for Epoch Tracker With Redemption", () => {
 
         try {
             // We will trigger a successful call to return the value set in the cache after the failed joinSession call
-            epochCallback.setCallback(async () => mockFetch({}, async () => {
-                return epochTracker.fetchFromCache(cacheEntry1, undefined, "other");
-            }, okResponse, false, false));
+            epochCallback.setCallback(async () => epochTracker.fetchFromCache(cacheEntry1, undefined, "other"));
 
             // Initial joinSession call will return 404 but after the timeout, the call will be retried and succeed
-            await mockFetch({ headers: { "x-fluid-epoch": "epoch1" } }, async () => {
```

Comment:
Note - code before was confusing first argument as being actual payload of response (in successful case), as well as Response (as in containing headers, in failed case). This was very confusing, especially given headers is valid thing to have on successful path.

So I broke it into two separate arguments (1st and 3rd) and made more generic mockFetchCore that does not even operate with these concepts (it just takes callback that returns whole Response).

Reply:
I really like this separation of concerns between the two calls